book/src/libraries/external_functions.md

```diff
@@ -55,63 +60,79 @@ could be [variadic functions](https://en.cppreference.com/w/c/variadic).
 These functions are usually defined with the last parameter `...`, and signify
 that a function can be called with unlimited parameters.
 
-An example of a variadic function is printf.
+An example of a variadic function is `printf`.
 
 Calling a variadic function is supported in `ST`. To mark an external function
-as variadic, you can add a parameter of type `...` to the `VAR_INPUT` block
+as variadic, you can add a parameter of type `...` to the `VAR_INPUT` block.
 
-#### Variadic Function Example
+#### Variadic function example
 
 Given the `printf` function defined as
 
 ```C
 int printf( const char *restrict format, ... );
 ```
 
-The `ST` interface can be defined as
+the `ST` interface can be defined as
 
 ```iecst
-@EXTERNAL
+{external}
 FUNCTION printf : DINT
-VAR_INPUT
+VAR_IN_OUT
   format : STRING;
+END_VAR
+VAR_INPUT
   args : ...;
 END_VAR
 END_FUNCTION
 ```
 
-#### Full Example
+#### Runnable example
 
-With the printf function available on the system, there is no need to declare
-the C function
+With the `printf` function available on the system, there is no need to declare
+the C function.
 
-Declare an `ST` program called `ExternalFunctions.st` with the following code:
+An `ST` program called `ExternalFunctions.st` with the following code can be declared:
 
 ```iecst
-@EXTERNAL FUNCTION printf : DINT
-VAR_INPUT
+(*ExternalFunctions.st*)
+
+(**
+ * The printf function's interface, marked as external since 
+ * it is defined directly along other ST functions
+ *)
+{external}
+FUNCTION printf : DINT
+VAR_IN_OUT
     format : STRING;
+END_VAR
+VAR_INPUT
     args: ...;
 END_VAR
 END_FUNCTION
 
 
 (**
 * The main function of the program prints a demo to the standard out
+* The function main is implemented at this location and thus not marked
+* as {external}
 *)
 FUNCTION main : DINT
   main := 1;
   printf('Value %d, %d, %d', main, main * 10, main * 100);
```

Comment:
I tried the example on this branch, and unfortunately it does not work.

codegen cannot find the reference `main` in the variadic parmeter list :-(
```bash
error: SyntaxError { message: "no type hint available for Reference {\n    name: \"main\",\n}", range: SourceRange { range: 347..351, file: Some("my_printf.st") }, err_no: codegen__general }
``` 
when i introduce a variable, it works - this is probably its own issue, but lets add an example that works with this commit:
```st
FUNCTION main : DINT
  VAR foo : DINT; END_VAR
  
  foo := 1;
  printf('Value %d, %d, %d', foo, foo * 10, foo * 100);
  main := foo;
END_FUNCTION
```